builtin/audit/file/backend_test.go

```diff
@@ -0,0 +1,75 @@
+package file
+
+import (
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/vault/audit"
+	"github.com/hashicorp/vault/helper/salt"
+)
+
+func TestAuditFile_fileModeNew(t *testing.T) {
+	salter, _ := salt.NewSalt(nil, nil)
+
+	modeStr := "0777"
+	mode, err := strconv.ParseUint(modeStr, 8, 32)
+
+	file := "auditTest.txt"
+
+	config := map[string]string{
+		"path": file,
+		"mode": modeStr,
+	}
+
+	_, err = Factory(&audit.BackendConfig{
+		Salt:   salter,
+		Config: config,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(file)
+
+	info, _ := os.Stat(file)
+	createdMode := info.Mode()
+	if createdMode != os.FileMode(mode) {
+		t.Fatalf("File Mode does not match.")
+	}
+}
+
+func TestAuditFile_fileModeExisting(t *testing.T) {
+	salter, _ := salt.NewSalt(nil, nil)
+
+	mode := os.FileMode(0600)
```

Comment:
It's strange to have this declared up here when it's only used in a single test at the end -- just test against the expected value directly.